src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3017,6 +3151,21 @@ internal uint GetBroadestValEscape(BoundTupleExpression expr, uint scopeOfTheCon
             return broadest;
         }
 
+#nullable enable 
+
+        internal uint GetValEscape(BoundExpression expr, Symbol? symbol, uint scopeOfTheContainingExpression)
```

Comment:
This feels wrong but couldn't think of a better way to do it. The other way I considered was looking at the current method under the binder, if it's a constructor and we're looking at `BoundThisReference` then it's the equivalent check. That would avoid making a new method. Wasn't sure if that would catch all cases though.